pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -1155,22 +1156,39 @@ type cutoverProvider interface {
 // custoverFromJobProgress is a cutoverProvider that decides whether the cutover
 // time has been reached based on the progress stored on the job record.
 type cutoverFromJobProgress struct {
-	registry *jobs.Registry
-	jobID    jobspb.JobID
+	db    isql.DB
+	jobID jobspb.JobID
 }
 
 func (c *cutoverFromJobProgress) cutoverReached(ctx context.Context) (bool, error) {
-	j, err := c.registry.LoadJob(ctx, c.jobID)
-	if err != nil {
+	var (
+		progressBytes []byte
+		exists        bool
+	)
+	if err := c.db.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+		infoStorage := jobs.InfoStorageForJob(txn, c.jobID)
+		var err error
+		progressBytes, exists, err = infoStorage.GetLegacyProgress(ctx)
```

Comment:
any reason you didn't also refactor the stream_ingestion_frontier_processor to also use `GetLegacyProgress` and `GetLegacyPayload`?

Reply:
In the frontier processor, we want to touch the jobs table since that is where the claim data lives and we want to ensure the claim is still valid as part of the update.  But, the way I've done it in this draft is actually a bit of a bummer since it'll end up loading the job twice on every update. I should probably grab the claim data once and then keep the job object around.